gnmi/gnmi.go

```diff
@@ -260,6 +281,30 @@ func updateCacheNotifs(cache *cache.Cache, nos []*gpb.Notification, target, orig
 // the gNMI cache.
 // - auth adds authorization to before writing vals to the cache, if set to nil, not authorization is checked.
 func set(schema *ytypes.Schema, cache *cache.Cache, target string, req *gpb.SetRequest, preferShadowPath bool, validators []func(*oc.Root) error, timestamp int64, user string, auth PathAuth) error {
+	// skip validation and diffing for performance.
+	// TODO(wenbli): This if block should be removed once a more performant diffing utility is created.
+	if !preferShadowPath && len(req.Delete)+len(req.Replace) == 0 {
```

Comment:
i thought we didn't need to validate state paths at all?

Reply:
replace/delete don't work because they affect the config paths as well. The longterm solution is needed for those cases.